### PR TITLE
Final touch

### PR DESCRIPTION
**Pull request by SebChw**
Most commits are hopefully self.explanatory. But for these that are not:
1. [fix hashing problem (LighntingModule Compatibility)](https://github.com/SebChw/Actually-Robust-Training/commit/8a1c8a5092c8303adbcc0aa49ddff690fe8537db) We were talking today about availability to use ART with LightningModule. By default it won't have get_hash function
2. [fix: changed model not always added](https://github.com/SebChw/Actually-Robust-Training/commit/e181df670633e64061ee2a6ad7e0a4b64264300c) Idk why `False` was returned all the time and steps that was changed and succesfully passed check was not added. Moreover this can be done directly during checking checks

### CODE REVIEW

File: art/dashboard/backend.py [hunk 1]
```diff
@@ -37,7 +37,7 @@ def prepare_steps_info(logs_path: Path) -> Dict[str, Dict]:
                     "timestamp": run["timestamp"],
                     "hash": run["hash"],
                     "commit_id": run["commit_id"],
-                    "successfull": run["succesfull"],
+                    "successfull": run["successfull"],
```
Comment (kordc) [inline art/dashboard/backend.py]: So close, just remove the last "L", there's only one at the and actually

File: art/dashboard/app.py [hunk 2]
```diff
@@ -22,8 +22,8 @@
 if not LOGS_PATH.exists():
     raise ValueError(f"Path {LOGS_PATH} does not exist.")
 app = Dash(__name__, external_stylesheets=[dbc.themes.BOOTSTRAP])
-ORDERED_STEPS = prepare_steps(LOGS_PATH)
 STEPS_INFO = prepare_steps_info(LOGS_PATH)
+ORDERED_STEPS = [x for x in prepare_steps() if x in STEPS_INFO.keys()]
```
Comment (kordc) [inline art/dashboard/app.py:26]: If these changes are applied I would change the name STEPS_INFO to existing steps or something, that differentiate ORDERED_STEPS from STEPS_INFO - first are actually the hardcoded ones, and the other are the existing ones. For now the names suggest we're talking about the same steps, but with and without order

File: art/dashboard/backend.py (hunk 1 above)
Comment (SebChw) [reply art/dashboard/backend.py]: xD

Comment (SebChw) [reply art/dashboard/backend.py]: you are right

Comment (SebChw) [reply art/dashboard/backend.py]: xD

File: art/dashboard/app.py (hunk 2 above)
Comment (SebChw) [reply art/dashboard/app.py:26]: now ORDERED_STEPS are only the one that exists within STEPS_INFO

**Review by kordc — DISMISSED** [review]

**Pull request merged**